app/pages/ResetPasswordPage.js

```diff
@@ -0,0 +1,159 @@
+
+  render() {
+    const { user, model, form, handleSubmit } = this.props
+    return (
+      <RequestResetEmailForm
+        model={model}
+        form={form}
+        user={user}
+        submitClicked={this.onSubmitClicked}
+        handleSubmit={handleSubmit}
+      />
+    )
+  }
+}
+
+RequestResetEmail.propTypes = {
+  user: PropTypes.shape({
+    resetPasswordEmailFailure: PropTypes.bool,
+    resetPasswordEmailSuccess: PropTypes.bool
+  }).isRequired,
+  form: PropTypes.object.isRequired,
+  model: PropTypes.string.isRequired
+}
+
+const mapStateToProps = state => {
+  return {
+    ...formProps(state, 'resetPasswordEmailForm'),
+    user: state.user
+  }
+}
+
+const mapDispatchToProps = dispatch => {
+  return {
+    handleSubmit: model => dispatch(sendResetPasswordEmail(model))
+  }
+}
+
+export const RequestResetEmailContainer = withRouter(connect(mapStateToProps, mapDispatchToProps)(RequestResetEmail))
+
+class ResetPassword extends BaseForm {
+  static propTypes = {
+    action: PropTypes.string,
+    csrf_token: PropTypes.string,
+    form: PropTypes.object.isRequired,
+    errors: PropTypes.object
+  }
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      submitClicked: null
+    }
+    this.handleSubmit = this.handleSubmit.bind(this)
+  }
+
+  componentWillMount() {
+    let tokenString = this.props.match.params.tokenString
+    this.props.loadInitialData(tokenString)
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (this.props.form.validity.passwordsMatch !== nextProps.form.validity.passwordsMatch) {
+      if (!nextProps.form.validity.passwordsMatch) {
+        this.props.passwordsMatchMessage('Passwords do not match.')
+      }
+    }
+  }
+
+  handleSubmit(values) {
+    const { user } = this.props
+    const payload = Object.assign({}, values, user.user)
+    this.props.handleSubmit(payload)
+  }
+
+  onSubmitClicked = () => {
+    this.setState({
+      submitClicked: new Date().valueOf()
+    })
+  }
+
+  render() {
+    const { user, model, form } = this.props
+    return (
+      <ResetPasswordForm
+        model={model}
+        form={form}
+        user={user}
+        submitClicked={this.onSubmitClicked}
+        handleSubmit={this.handleSubmit}
+      />
+    )
+  }
+}
+
+ResetPassword.propTypes = {
+  user: PropTypes.shape({
+    resetPasswordSuccess: PropTypes.bool
+  }).isRequired,
+  form: PropTypes.object.isRequired,
+  model: PropTypes.string.isRequired
+}
+
+const mapResetStateToProps = state => {
+  return {
+    ...formProps(state, 'resetPasswordForm'),
+    user: state.user
+  }
+}
+
+const mapResetDispatchToProps = dispatch => {
+  return {
+    handleSubmit: payload => dispatch(submitResetPassword(payload)),
+    loadInitialData: token => dispatch(getUserDataFromToken(token)),
+    passwordsMatchMessage: message => dispatch(setErrorMessage(message))
+  }
+}
+
+export const ResetPasswordContainer = withRouter(connect(mapResetStateToProps, mapResetDispatchToProps)(ResetPassword))
```

Comment:
I don't get where this is supposed to fit in - shouldn't it be part of the page?